lib/rspec/core/example.rb

```diff
@@ -124,6 +124,22 @@ def self.parse_id(id)
         id.match(/\A(.*?)(?:\[([\d\s:,]+)\])?\z/).captures
       end
 
+      # Duplicates the example and overrides metadata with the provided
+      # hash.
+      #
+      # @param metadata_overrides [Hash] the hash to override the example metadata
+      # @return [Example] a duplicate of the example with modified metadata
+      def duplicate_with(metadata_overrides={})
+        new_metadata = metadata.clone.merge(metadata_overrides)
+
+        RSpec::Core::Metadata::RESERVED_KEYS.each do |reserved_key|
+          new_metadata.delete reserved_key
+        end
+
+        Example.new(example_group.clone, description.clone,
```

Comment:
The example group should not be cloned.  The example should belong to the _same_ example group class, not a copy of it. The clone is likely to cause some bugs.